thoth/messaging/message_factory.py

```diff
@@ -19,13 +19,13 @@
 """This is Thoth Messaging module for message_factory."""
 
 import faust
-from typing import Tuple
+import typing
```

Comment:
> One final note: when adding types, the convention is to import types using the form `from typing import Iterable` (as opposed to doing just `import typing` or `import typing as t` or `from typing import *`).


https://mypy.readthedocs.io/en/stable/getting_started.html#the-typing-module

Reply:
The reason we import typing here is to allow more flexibility in terms of what they user can pass as message types.  So that they can pass something like `[("name", "str"), ("meta", "typing.Dict[str, typing.Any])]`

Reply:
You mean for `message_contents`? Is the typing import necessary in that case?

Reply:
Yes, because we use `exec()` it is exactly the same as:
```
MessageContents(...):
   a: str
   b: typing.Dict(str, typing.Any)
```
and it needs any classes used in the type annotation to be present upon execution.